src/renderers/WebGLRenderer.js

```diff
@@ -1853,15 +1854,60 @@ function WebGLRenderer( parameters = {} ) {
 
 	};
 
-	this.setRenderTarget = function ( renderTarget, activeCubeFace = 0, activeMipmapLevel = 0 ) {
+	this.setRenderTarget = function ( renderTarget, activeCubeFace = 0, activeMipmapLevel = 0, options = {} ) {
 
 		_currentRenderTarget = renderTarget;
 		_currentActiveCubeFace = activeCubeFace;
 		_currentActiveMipmapLevel = activeMipmapLevel;
+		const useDefaultFramebuffer = options.framebuffer === undefined;
 
-		if ( renderTarget && properties.get( renderTarget ).__webglFramebuffer === undefined ) {
+		if ( renderTarget ) {
+
+			const renderTargetProperties = properties.get( renderTarget );
+			let hasNewExternalTextures = false;
+
+			if ( options.colorTexture !== undefined ) {
+
+				hasNewExternalTextures = true;
+				properties.get( renderTarget.texture ).__webglTexture = options.colorTexture;
+
+				renderTarget.autoAllocateDepthBuffer = options.depthTexture === undefined;
+
+				if ( ! renderTarget.autoAllocateDepthBuffer ) {
 
-			textures.setupRenderTarget( renderTarget );
+					properties.get( renderTarget.depthTexture ).__webglTexture = options.depthTexture;
+
+					// The multisample_render_to_texture extension doesn't work properly if there
+					// are midframe flushes and an external depth buffer. Disable use of the extension.
+					if ( renderTarget.useMultisampleRenderToTexture ) {
```

Comment:
Maybe worth adding a log/warning here. How likely is it that you would end up in this state without user error?

Reply:
It's not likely, but the author can force a texture if they wanted to. 
I'll add a warning.